locust/web.py

```diff
@@ -36,17 +39,45 @@
 
 class WebUI:
     server = None
-    """Refernce to pyqsgi.WSGIServer once it's started"""
+    """Reference to pyqsgi.WSGIServer once it's started"""
     
-    def __init__(self, environment):
+    def __init__(self, environment, web_auth=None):
         environment.web_ui = self
         self.environment = environment
         app = Flask(__name__)
         self.app = app
         app.debug = True
         app.root_path = os.path.dirname(os.path.abspath(__file__))
+        self.app.config["BASIC_AUTH_ENABLED"] = False
+        self.auth = None
+
+        if web_auth is not None:
+            credentials = web_auth.split(':')
+            if len(credentials) == 2:
+                self.app.config["BASIC_AUTH_USERNAME"] = credentials[0]
+                self.app.config["BASIC_AUTH_PASSWORD"] = credentials[1]
+                self.app.config["BASIC_AUTH_ENABLED"] = True
+                self.auth = BasicAuth()
+                self.auth.init_app(self.app)
+            else:
+                sys.stderr.write("The credentials need to provided in the format username:password\n")
+                sys.exit(1)
```

Comment:
Instead of writing to stderr and calling `sys.exit()` we should raise an exception here. For example `InvalidCredentialsError` that could inherit from `ValueError`. Then we can catch this exception in main where we can output a proper error message with `logger.error()` and call `sys.exit(1)`.

Reply:
We're still not raising an exception here. 

We shouldn't catch ValueError and IndexError (that could cause us to mistakingly swallow unrelated errors in BasicAuth). What I meant was that instead of calling `sys.exit` *here* we should raise our *own* exception (that can  inherit from ValueError) and then handle this exception in main.py where we can log the error and call `sys.exit(1)`. I see no problem with the `if len(credentials) == 2:` check.

Reply:
Okay I've had another attempt at this, let me know you're happy with it.

Reply:
Looks great! Only some minor fixes that I could fix from the Github UI.